go-controller/pkg/ovn/controller/egressservice/egressservice_zone.go

```diff
@@ -451,51 +460,60 @@ func (c *Controller) repair() error {
 	}
 
 	if config.OVNKubernetesFeature.EnableInterconnect {
-		lrsrPredicate := func(item *nbdb.LogicalRouterStaticRoute) bool {
+		lrpICPredicate := func(item *nbdb.LogicalRouterPolicy) bool {
+			if item.Priority != ovntypes.EgressSVCReroutePriority {
+				return false
+			}
 			svcKey, found := item.ExternalIDs[svcExternalIDKey]
 			if !found {
 				return false
 			}
 
-			if item.Policy == nil || *item.Policy != nbdb.LogicalRouterStaticRoutePolicySrcIP {
-				klog.Infof("Egress service repair will delete lrsr for service %s it has an invalid policy: %v", svcKey, item)
-				return true
-			}
-
 			svc, found := c.services[svcKey]
+			if found {
+				klog.Infof("Egress service repair already finished repairing lrp for service %s: %v", svcKey, item)
+				return false
+			}
 			if !found {
-				klog.Infof("Egress service repair will delete lrsr for service %s because it is no longer a valid egress service: %v", svcKey, item)
-				return true
+				trimmedKeyValue := strings.TrimSuffix(svcKey, "-ic")
```

Comment:
this "-ic" suffix...should it be a constant somewhere?

Reply:
I'm guessing you add this -ic policy cause there is already another policy in egress service. Maybe some explanation in a comment or the commit message to see how they look and interact would be good.

Reply:
yea let me add `-ic` as a constant, and yes you are absolutely right that this is confusing. Let me explain this a bit better.
There are two policies that are created by ESVC. 

1. One is on the node on which the pod lives which redirects the traffic to the remote transit switchIP *if the egressNode != pod.Node*
2. Second is the new policy that I've added (used to be LRSR) which is done on the egressNode for any remote pod it controls to reroute traffic to its mp0. So basically it receives traffic forwarded by the above 1st policy and shoves it into the mp0 of this node.

I had to add the `-ic` for the second policy case - only for this specific repair cleanup we do in the beginning because the logic to cleanup the two types of policies is different and for nonIC we don't need to bother with this second type, so I added the `-ic` as a differentiator. 

I will add more comments here to clarify this logic, good catch.

Reply:
done.